# Estimation Assessment
## Physics 211 - Bruce Emerson
### January 25, 2021
#### Zachary Bochanski

Estimation of the size of a mason jar, size of a coffee bean, number of total beans and density of beans in centimeters.

Order of operations: 
- Take care of all the constants required
- Estimate bean size
- Find jar dimensions
- Find jar volume
- Find bean volume
- Find beans per volume

### Imports, constants

numpy package for math operations.

JAR_RADIUS constant for jar radius cm
<br/>
JAR_HEIGHT constant for jar height cm 
<br/>
BEAN_LENGTH constant for bean length cm
<br/>
BEAN_WIDTH constant for bean width cm
<br/>
BEAN_HEIGHT constant for bean height cm
<br/>
PI constant for $ \pi $ 


In [13]:
import numpy as np

JAR_RADIUS = None
JAR_HEIGHT = None
BEAN_LENGTH = None
BEAN_WIDTH = None
BEAN_HEIGHT = None
PI = np.pi

### Part I: Unit sizes

First thing is taking a look at the size of a bean. The bean size becomes a unit for determining measurement totals and from there calculations made about jar size, bean density, bean volume etc.

Looking at one bean, as there happens to be a bag open from this morning, they are about one finger-nail long and the width and height are about half. 

One bean's dimenshions look to be about:

$ bean=.4cm\: x\: .5cm\: x\: 1cm\: $

1 bean_length = 1cm

In [14]:
BEAN_LENGTH = 1
BEAN_WIDTH = .5
BEAN_HEIGHT = .4

### Part II: Jar size in bean units

Now that average bean size has been estimated the `bean unit` can be applied to the image of the jar.

Looks like this jar has a diameter of about `8 bean_lengths` and the height is `12 bean_lengths` which it has been determined 1 bean_length is *1cm*:

![bean_diameter](https://github.com/ztbochanski/physics-211/raw/main/images/bean_jar.jpg)


In [15]:
JAR_RADIUS = 8
JAR_HEIGHT = 12

### Part III: Finding Jar Volume

The equation for a cylinder:

$ V = \pi r^2h $

In [20]:
jar_volume = PI*(JAR_RADIUS**2)*JAR_HEIGHT
print("There are", format(jar_volume, ".1f"), "cubic centimeters of jar space")

There are 2412.7 cubic centimeters of jar space


### Part IV: Single bean volume

The beans dimensions as determined in part I in cm:

$ bean=.4cm\: x\: .5cm\: x\: 1cm\: $

In [22]:
bean_volume = .4*.5*1
print("One bean is about", format(bean_volume, ".1f"), "cubic centimeters") 

One bean is about 0.2 cubic centimeters


### Part V: Beans per cm

At .2 cubic centimerters per bean how many can fit in one cubic centimeter:

In [23]:
beans_per_cubiccm = 1/.5 # 10/2 = 5 beans per cm3
print(beans_per_cubiccm, "beans per cm3")

2.0 beans per cm3


### Part VI: Bean space occupied (volume)

The beans are almost the height of the jar, looks like the height is at 11 bean_lenghs = 11cm

In [21]:
bean_space_occupied = PI*(JAR_RADIUS**2)*11
print("There are", format(bean_space_occupied, ".1f"), "cubic centimeters of coffee beans")

There are 2211.7 cubic centimeters of coffee beans


### Part VII: Estimated number of beans

It is estimated that there are 2 beans per centimeteter3 and 2211.7 centimeteters of bean space in the jar.

How many beans? Bean space divided by beans per centimeter.

In [27]:
number_of_beans = bean_space_occupied/beans_per_cubiccm
print("It seems like there are", format(number_of_beans, ".0f"), "beans in the jar")

It seems like there are 1106 beans in the jar


### Discussion

This seems like a reasonable estimate maybe a little on the high side, the bean units were pretty generous in length and went a bit past usable jar space. The thing is I can be pretty sure that it is an overestimate. It's difficult to visualize volumes in general especial cubic centimeters. My inital reaction was 2000cm3 of beans seems a bit too high! but 2 beans per cm3 seems reasonable, and 1000 beans in a jar seems reasonable.